src/layouts/ProfileLayout.tsx

```diff
@@ -45,69 +51,94 @@ function ProfileLayout({ children, wallet }: ProfileLayout): JSX.Element {
         />
         <link rel="icon" href="/favicon.ico" />
       </Head>
-      <Overview>
-        <Overview.Hero>
-          <Overview.Info
-            avatar={<Overview.Avatar src={wallet.previewImage} circle />}
-            title={<Overview.Title>{wallet.displayName}</Overview.Title>}
-          >
-            <Overview.Actions>
-              {isFollowingProfile ? (
+      <WalletIdentityProvider appName="Holaplex" appTwitter="@holaplex">
+        <Overview>
+          <Overview.Hero>
+            <Overview.Info
+              avatar={<Overview.Avatar src={wallet.previewImage} circle />}
+              title={<Overview.Title>{wallet.displayName}</Overview.Title>}
+            >
+              <Overview.Actions>
+                {isMyProfile && !wallet.profile?.handle && anchorWallet && (
```

Comment:
```suggestion
const { canConnect } = useConnectTwitter(address)
                {canConnect && (
```
Can read the viewer reactive var from apollo to get the if profile exists already and connected status. The viewer won't be available unless the user connects.